_data-prepper/pipelines/configuration/processors/mutate-strings.md

```diff
@@ -0,0 +1,237 @@
+
+Mutate string processesors allow you to change the way that a string appears. For example, you can use the `uppercase_string` processor to convert a string to uppercase, and you can use the `lowercase_string` processor to convert a string to lowercase. The following is a list of processors that allow you to mutate a string:
+
+* [substitute_string](#substitutestringprocessor)
+* [split_string](#splitstringprocessor)
+* [uppercase_string](#uppercasestringprocessor)
+* [lowercase_string](#lowercasestringprocessor)
+* [trim_string](#trimstringprocessor)
+
+## substitute_string
+
+The `substitute_string` processor matches a key's value against a regular expression (regex) and replaces all matches with a replacement string.
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - substitute_string:
+        entries:
+          - source: "message"
+            from: ":"
+            to: "-"
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` of the file source in your `pipeline.yaml` file with your file path:
+
+```json
+{"message": "ab:cd:ab:cd"}
+```
+
+When you run Data Prepper with the `pipeline.yaml` file, you will see the following output:
+
+```json
+{"message": "ab-cd-ab-cd"}
+```
+
+If the `from` regex string does not return a match, the key returns without any changes.
+
+### Configuration
+
+<!--- Are these paramters? How are these items used? Need some intro text here.--->
+
+* `entries` (required): A list of entries to add to an event.
+    * `source` (required): The key to be modified.
+    * `from` (required): The regex string to be replaced. Special regex characters such as `[` and `]` must be escaped using `\\` when using double quotes and `\` when using single quotes. See [Class Pattern](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Pattern.html) for more information.
+    * `to` (required): The string to be substituted for each match of `from`.
+    
+## split_string
+
+The `split_string` processor splits a field into an array using a delimiter character.
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - split_string:
+        entries:
+          - source: "message"
+            delimiter: ","
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with your file path:
+
+```json
+{"message": "hello,world"}
+```
+When you run Data Prepper with this `pipeline.yaml` file, you should see the following output:
+
+```json
+{"message":["hello","world"]}
+```
+
+### Configuration
```

Comment:
Same as previous Configuration sections: What does configuration specifically allow the user to do here? What are the following items referred to as? Parameters?